backend/src/services/base.service.ts

```diff
@@ -98,4 +105,30 @@ export abstract class BaseService<T, C, U> {
   async getAll(): Promise<T[]> {
     return this.model.find();
   }
+
+  /**
+   * Appends the uploaded file to a document with given ID.
+   *
+   * @param _file - The file to append.
+   * @param _documentID - The ID of the document to which to append the file
+   * @param _fileType - Type of the uploaded file.
+   * @returns A promise that resolves to the updated entity.
+   */
+  abstract appendFile(
+    _file: Express.Multer.File,
+    _documentID: MongooseObjectId,
+    _fileType: SupportedRawFileTypes
+  ): Promise<T>;
+
+  /**
+   * Creates all datafiles from the uploaded dataset file
+   *
+   * @param _file - The dataset file to use.
+   * @param _dataset - The dataset type of the file.
+   * @returns A promise that resolves to all created entities.
+   */
+  abstract datasetFile(
```

Comment:
should be also inside a interface or thb maybe we can drop the declariation.
inside a base service must be only logic that 100 % are shared across all services.

Reply:
Done.